### PR TITLE
fix(header): optimize large title

### PR DESCRIPTION
**Pull request by liamdebeasi**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Large title needed some optimizations

## What is the new behavior?

- Collapse is now reactive
- Added RTL support
- Cleaned up old code
- Fixed types
- Added E2E tests
- Added usage docs

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: core/src/components/buttons/readme.md [hunk 1]
```diff
@@ -55,6 +55,15 @@ The `<ion-buttons>` element can be positioned inside of the toolbar using a name
...
   </ion-buttons>
 </ion-toolbar>
+
+<ion-toolbar>
+  <ion-buttons collapse="true">
+    <ion-button>
+      <ion-icon slot="icon-only" name="star"></ion-icon>
+    </ion-button>
+  </ion-buttons>
+  <ion-title>Collapsable Buttons</ion-title>
```
Comment (brandyscarney) [inline core/src/components/buttons/readme.md]:
```suggestion
  <ion-title>Collapsible Buttons</ion-title>
```

File: core/src/components/buttons/readme.md [hunk 2]
```diff
@@ -100,6 +109,15 @@ The `<ion-buttons>` element can be positioned inside of the toolbar using a name
...
```
Comment (brandyscarney) [inline core/src/components/buttons/readme.md]:
```suggestion
  <ion-title>Collapsible Buttons</ion-title>
```

File: core/src/components/buttons/readme.md [hunk 3]
```diff
@@ -155,6 +173,15 @@ export const ButtonsExample: React.FC = () => (
...
       </IonButtons>
     </IonToolbar>
+    
+    <IonToolbar>
+      <IonButtons collapse="true">
+        <IonButton>
+          <IonIcon slot="icon-only" name="star" />
+        </IonButton>
+      </IonButtons>
+      <IonTitle>Collapsable Buttons</IonTitle>
```
Comment (brandyscarney) [inline core/src/components/buttons/readme.md]:
```suggestion
      <IonTitle>Collapsible Buttons</IonTitle>
```

File: core/src/components/buttons/readme.md [hunk 4]
```diff
@@ -199,16 +226,25 @@ export const ButtonsExample: React.FC = () => (
...
     </ion-buttons>
   </ion-toolbar>
+  
+  <ion-toolbar>
+    <ion-buttons collapse="true">
+      <ion-button>
+        <ion-icon slot="icon-only" name="star"></ion-icon>
+      </ion-button>
+    </ion-buttons>
+    <ion-title>Collapsable Buttons</ion-title>
```
Comment (brandyscarney) [inline core/src/components/buttons/readme.md]:
```suggestion
    <ion-title>Collapsible Buttons</ion-title>
```

File: core/src/components/buttons/usage/angular.md [hunk 5]
```diff
@@ -34,4 +34,13 @@
...
   </ion-buttons>
 </ion-toolbar>
+
+<ion-toolbar>
+  <ion-buttons collapse="true">
+    <ion-button>
+      <ion-icon slot="icon-only" name="star"></ion-icon>
+    </ion-button>
+  </ion-buttons>
+  <ion-title>Collapsable Buttons</ion-title>
```
Comment (brandyscarney) [inline core/src/components/buttons/usage/angular.md]:
```suggestion
  <ion-title>Collapsible Buttons</ion-title>
```

File: core/src/components/buttons/usage/javascript.md [hunk 6]
```diff
@@ -38,4 +38,13 @@
...
   </ion-buttons>
 </ion-toolbar>
+
+<ion-toolbar>
+  <ion-buttons collapse="true">
+    <ion-button>
+      <ion-icon slot="icon-only" name="star"></ion-icon>
+    </ion-button>
+  </ion-buttons>
+  <ion-title>Collapsable Buttons</ion-title>
```
Comment (brandyscarney) [inline core/src/components/buttons/usage/javascript.md]:
```suggestion
  <ion-title>Collapsible Buttons</ion-title>
```

File: core/src/components/buttons/usage/react.md [hunk 7]
```diff
@@ -48,6 +48,15 @@ export const ButtonsExample: React.FC = () => (
...
       </IonButtons>
     </IonToolbar>
+    
+    <IonToolbar>
+      <IonButtons collapse="true">
+        <IonButton>
+          <IonIcon slot="icon-only" name="star" />
+        </IonButton>
+      </IonButtons>
+      <IonTitle>Collapsable Buttons</IonTitle>
```
Comment (brandyscarney) [inline core/src/components/buttons/usage/react.md]:
```suggestion
      <IonTitle>Collapsible Buttons</IonTitle>
```

File: core/src/components/buttons/usage/vue.md [hunk 8]
```diff
@@ -35,5 +35,14 @@
...
     </ion-buttons>
   </ion-toolbar>
+  
+  <ion-toolbar>
+    <ion-buttons collapse="true">
+      <ion-button>
+        <ion-icon slot="icon-only" name="star"></ion-icon>
+      </ion-button>
+    </ion-buttons>
+    <ion-title>Collapsable Buttons</ion-title>
```
Comment (brandyscarney) [inline core/src/components/buttons/usage/vue.md]:
```suggestion
    <ion-title>Collapsible Buttons</ion-title>
```

File: core/src/components/header/header.tsx [hunk 9]
```diff
@@ -15,14 +15,18 @@ import { cloneElement, createHeaderIndex, handleContentScroll, handleToolbarInte
...
   private contentScrollCallback?: any;
+  private intersectionObserver?: any;
 
   @Element() el!: HTMLElement;
 
   /**
    * If `true`, the header will collapse on scroll of the content.
    * Only applies in `ios` mode.
+   *
+   * Typically used for [Collapsable Large Titles](https://ionicframework.com/docs/api/title#collapsable-large-titles)
```
Comment (brandyscarney) [inline core/src/components/header/header.tsx]:
```suggestion
   * Typically used for [Collapsible Large Titles](https://ionicframework.com/docs/api/title#collapsible-large-titles)
```

**Comment by liamdebeasi** [comment]
After typing collapsable so many time I'm surprised I didn't mispell it more often 😂

File: core/src/components/title/readme.md [hunk 10]
```diff
@@ -2,23 +2,124 @@
...
+
+```html
+<ion-header>
+  <ion-toolbar>    
+    <ion-title>Settings</ion-title>               
+  </ion-toolbar>
+</ion-header>
+
+<ion-content>
+  <ion-header collapse="true>
```
Comment (brandyscarney) [inline core/src/components/title/readme.md]:
```suggestion
  <ion-header collapse="true">              
```

File: core/src/components/title/readme.md [hunk 11]
```diff
@@ -2,23 +2,124 @@
...
+  <ion-toolbar>   
+    <ion-buttons collapse="true">
+      <ion-button>Click Me</ion-button>
+    </ion-buttons> 
+    <ion-title>Settings</ion-title>               
+  </ion-toolbar>
+</ion-header>
+
+<ion-content>
+  <ion-header collapse="true>
```
Comment (brandyscarney) [inline core/src/components/title/readme.md]:
```suggestion
  <ion-header collapse="true">              
```

**Pull request merged**